CalPatRec/src/PhiClusterFinder_module.cc

```diff
@@ -0,0 +1,566 @@
+            }
+          }
+        }
+        //  Alternate case when the hits are around 0 and 2PI
+        else if(cluphimax < cluphimin){
+          for(int i=0; i<nh; i++){
+            const mu2e::ComboHit* ch = &_chcol->at(ordchcol[i]);
+            Double_t phi = ch->phi();
+            if(phi < 0) phi += 2*M_PI;
+            if((phi>=cluphimin and phi< _phimax) or (phi<=cluphimax and phi>=0)){
+              finalcount++;
+              if(phi > M_PI) phi = phi - 2*M_PI;
+              usedhit[i] = true;
+              countusedhit++;
+              count2++;
+              phi1 = phi1+phi;
+              _clustno[i] = counter;
+              // if(_debug>3)std::cout<<"Alternate cluster case = "<<_clustno[i]<<" Cluster no. = "<<counter<<" hit = "<<i<<" Pos x = "<<ch->pos().X()<<" y = "<<ch->pos().Y()<<std::endl;
+           }
+         }
+        }
+        else {
+          // Hit associated to no cluster. Note : Need to investigate further
+          // printf("Min phi = %f Max phi = %f \n",cluphimin,cluphimax);
+          _clustno.clear();
+          break;
+        }
+        //Total phi and no. of hits in a cluster
+        if (phi1 !=0) {
+          _phitotal[counter] = phi1;
+          _nhitotal[counter] = finalcount;
+        }
+        // Increment the cluster number
+        counter += 1;
+      }
+    }
+    // phi separation between the clusters. Note : Only used for the events with two phi clusters at the moment
+    if (counter == 2) {
+      dphi = fabs(_phitotal[0]/_nhitotal[0]-_phitotal[1]/_nhitotal[1]);
+      if (dphi > M_PI) dphi = 2*M_PI-dphi;
+    }
+    // Events where two phi clusters are found but they are separated < min delta phi.
+    if (counter == 2 and dphi < _mindphi){
+      TimeCluster otc;
+      fillTimeSpectrumAlt(ordchcol);
+      findPeaks(otc, ordchcol);
+      assignHits(otc, ordchcol);
+      initCluster(otc);
+      if(otc._nsh > _minnsh) tccol1.push_back(otc);
+      if (_debug > 1) {
+        int onsh = otc._nsh;
+        printf("dphi = %f nh = %i nsh = %i \n",dphi,nh,onsh);
+      }
+    }
+    // Loop through the phi clusters and form time clusters
+    for(int j=0;j<counter;j++){
+       TimeCluster tc;
+       fillTimeSpectrum(j ,ordchcol);
+       findPeaks       (tc ,ordchcol);
+       for(int ih=0; ih<nh; ih++) {
+        //Fill the straw hit indices if the cluster number of the hit == j
+        if (_clustno[ih]==j) tc._strawHitIdxs.push_back(ordchcol[ih]);
+       }
+       initCluster(tc);
+       if (_debug > 1) {
+         int nsh = tc._nsh;
+         printf("Time cluster : %i No. straw hits : %i \n",counter,nsh);
+       }
+       // Sigma of the phi spectrum
+       float sigma(0);
+       if (tc._nsh > _minnsh) {
+         if (tc._nsh > _nphiclusters) {
+           sigma = checkdelta(tc, j, ordchcol);
```

Comment:
could you please change the name of the variable `sigma`? `::checkdelta` doesn't evaluate a standard deviation, so I think it's confusing. Alternatively, you can change the math exploited in that function, but then its threshold might need to be re-calibrated.